Add `AnimatableTableViewController` with designable `UIRefreshControl`

phimage opened the pull request:
and other features same as  `AnimatableViewController`

`RefreshControlDesignable` allow to custom the internal `UIRefreshControl`

`UIRefreshControl` must not be set in subview but using `refreshControl` attribute of `UITableViewController`
I think apple must do some job to make it accessible through Interface Builder, but until that this feature allow to custom the basic attribute of `UIRefreshControl`

---

I can go further to allow text edition
by adding something like that in protocol `RefreshControlDesignable`
```swift
var refreshControlTitle: String? { get set }
var refreshControlTitleFont: UIFont? { get set }
var refreshControlTitleColor: UIFont? { get set }
```
and there in configure function do the job to create an attributed string 
```swift

if let title = refreshControlTitle {
  var attributes: [String : Any] = [:]
  if let font = refreshControlTitleFont {
    attributes[NSFontAttributeName] = font
  }
  if let color = refreshControlTitleColor {
    attributes[NSForegroundColorAttributeName] = color
  }
  refreshControl?.attributedTitle = NSAttributedString(string: title, attributes: attributes)
} else {
  refreshControl?.attributedTitle = nil
} 
```

phimage committed:
Add `AnimatableTableViewController` with designable `UIRefreshControl`
and other features same as  `AnimatableViewController`

phimage committed:
Add `AnimatableTableViewController` with designable `UIRefreshControl`
and other features same as  `AnimatableViewController`

tbaranes reviewed (approved):
That `DesignableRefreshControl` is an Interesting idea!

Your PR looks good to me, just missing documentation stuff:
- A CHANGELOG entry
- A new section in `Documentation/APIs.md`

The title edition has definitely its place in this protocol, but we can already merged it in this state once the conflicts solved. We can add these features in another PR, it's up to you.

Also, once that one merged, it would be interesting to add an `AnimatableTableView`, `AnimatableCollectionView`, ... (any UI classes who may used a refresh control) and made them conform to `RefreshControlDesignable`.

tbaranes commented:
One more point: should we add an entry in the demo app ?

phimage committed:
Update CHANGELOG.md  and API.md for `AnimatableTableViewController` and `RefreshControlDesignable`
Update API.md for `SliderImagesDesignable`
Fix some API.md mistakes
Merge  'origin/master'

phimage commented:
CHANGELOG :white_check_mark:
Documentation/APIs.md :white_check_mark:
Merge conflict  :white_check_mark:

:construction: when trying to add into DemoApp.storyboard I have some weird issue
The colors are set, but there is logs  

> Failed to set (refreshControlTintColor) user defined inspected property on (IBAnimatable.AnimatableTableViewController): Could not load NIB in bundle: 'NSBundle </Users/phimage/Library/Developer/CoreSimulator/Devices/F34E4500-71DB-4F92-8BD1-FB0A0A5BFCA1/data/Containers/Bundle/Application/21B17425-4BA0-4FB8-B3DD-5216B5FF4E7E/IBAnimatableApp.app> (loaded)' with name 'DWs-Hm-hGO-view-ho8-uA-Rom'

I try with `AnimatableTableViewController` and with a new class in IBAnimatableApp named `TimeLineTableViewController` with code to simulate refresh
It's work yesterday on a new storyboard... (will try an another day with updated xcode)

JakeLin commented on CHANGELOG.md:
```diff
@@ -308,7 +310,7 @@ None
 - Add `Presenter` to manage Present and Dismiss transition animations
 - Add `PresentFadeSegue`, `PresentFadeInSegue` and `PresentFadeOutSegue` for Present transition with Fade animations
-- Add `PresentFadeWithDismissInteractionSegue`, `PresentFadeInWithDismissInteractionSegue` and `PresentFadeOutWithDismissInteractionSegue` for Present transition with Fade animations and getsture interactions.
+- Add `PresentFadeWithDismissInteractionSegue`, `PresentFadeInWithDismissInteractionSegue` and `PresentFadeOutWithDismissInteractionSegue` for Present transition with Fade animations and gesture interactions.
```
good on you for picking up all typos.

JakeLin commented on IBAnimatable/AnimatableTableViewController.swift:
```diff
@@ -0,0 +1,103 @@
+  @IBInspectable open var rootWindowBackgroundColor: UIColor?
+
+  // MARK: - TransitionAnimatable
+  @IBInspectable  var _transitionAnimationType: String? {
```
should be one space between `@IBInspectable  var` 😅

phimage replied:
well seen
copy paste from `AnimatableViewController`
Maybe the variable must also be `open` ?

tbaranes replied:
I think so, good catch

JakeLin commented on IBAnimatable/AnimatableTableViewController.swift:
```diff
@@ -0,0 +1,103 @@
+  open override func viewWillDisappear(_ animated: Bool) {
+    super.viewWillDisappear(animated)
+    resetHideNavigationBar()
+
```
remove the line 😉

phimage replied:
same here, blank line from `AnimatableViewController`

JakeLin reviewed (approved):
Such a great idea to make life easier. Well done 💪

JakeLin commented:
@phimage can you try to restart Xcode or delete derived data? It is very wired. If this issue still happens. You may change the name `refreshControlTintColor` to `refreshControlTintColorWhatever` to troubleshoot it. I don't have a straight answer on this; I think that maybe a Xcode bug 😢 It will be great to have a demo VC with this feature. Well done 💪

tbaranes commented:
Thanks for the update! It's perfect 💯 
Could you also rebased on master once you estimate that PR is ready to go?

JakeLin commented:
@phimage Could you please merge it when you think it is ready? For the Xcode error, @tbaranes had the same/similar issue on his machine because he had two version of Xcode (8.2 and 8.3). After he removed 8.2, everything works fine again.

Thanks again, well done 👍

e-marchand committed:
Use `AnimatableTableViewController` in demo app in time line view
Fix issue when view is not already loaded yet
Set open some IBInspectable variables in `AnimatableTableViewController` and `AnimatableViewController`

phimage committed:
Merge branch 'master' into feature-tablecontroller

phimage commented:
sorry some too much weird merge conflict I will start another one...
